mopidy/mpd/protocol/stored_playlists.py

```diff
@@ -16,6 +15,16 @@ def _check_playlist_name(name):
         raise exceptions.MpdInvalidPlaylistName()
 
 
+def _get_playlist(context, name, must_exist=True):
+    playlist = None
+    uri = context.lookup_playlist_uri_from_name(name)
+    if uri:
+        playlist = context.core.playlists.lookup(uri).get()
```

Comment:
Should we be using `mopidy.core.PlaylistsController.get_items` to reduce how much we lookup?

Feel free to postpone this for this PR.

Reply:
`core.playlist.lookup()` returns a `Playlist` object with full `Track` objects in its `Playlist.tracks` field. The core playlist API seems to be missing a way to get a playlist `Ref` with just the name of the playlist other than using `core.playlist.as_list()` and filtering the returned playlist `Ref`s by URI yourself. We should consider making an equivalent of `core.playlist.lookup()` to get a playlist `Ref`.

I see nothing wrong in the implementation of `_get_playlist()` in this PR. It makes nothing worse and reduces a bunch of duplication. Improving the core playlists API can come later.

:+1:

Reply:
I am completely confused regarding my original reply. I have no idea why I thought I was getting Refs from `core.PlaylistsController.lookup()`. @adamcik your suggestion of `core.PlaylistsController.get_items()` seems much better and I can only assume that's what I thought I was getting from `lookup()`! But then that doesn't fit with what I've done here since some of these routines do need the full `Track` objects and some don't. And this change makes `listplaylistinfo` lookup the `Track`s twice which is a bit weird.